From c29167165ef69e77193b2a9b0872cef49bfe06eb Mon Sep 17 00:00:00 2001
From: Pascal Paillet <p.paillet@st.com>
Date: Mon, 15 Jan 2024 15:16:21 +0100
Subject: [PATCH 0758/1695] dts: stm32: set LDO1 to VOUT4 input source on
 stm32mp257 boards

Set LDO1 to VOUT4 input source on stm32mp257 boards.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 1d6e5bd612ca437460ebc50e87e7feb845a92372 dts: stm32: set LDO1 to VOUT4 input source on stm32mp257 boards]
Change-Id: If847b711e82b56905cd2a7a0487ca22f7656e9b9
Signed-off-by: Pascal Paillet <p.paillet@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/373102
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
---
 core/arch/arm/dts/stm32mp257f-dk.dts  | 2 ++
 core/arch/arm/dts/stm32mp257f-ev1.dts | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/core/arch/arm/dts/stm32mp257f-dk.dts b/core/arch/arm/dts/stm32mp257f-dk.dts
index 31d0cdfd8..6f77d4e49 100644
--- a/core/arch/arm/dts/stm32mp257f-dk.dts
+++ b/core/arch/arm/dts/stm32mp257f-dk.dts
@@ -81,6 +81,7 @@
 		regulators {
 			compatible = "st,stpmic2-regulators";
 
+			ldo1-supply = <&vddio_pmic>;
 			vddcpu: buck1 {
 				regulator-name = "vddcpu";
 				regulator-min-microvolt = <800000>;
@@ -234,6 +235,7 @@
 				regulator-min-microvolt = <1800000>;
 				regulator-max-microvolt = <1800000>;
 				regulator-always-on;
+				st,alternate-input-source;
 				st,mask-reset;
 			};
 			vdd_emmc: ldo2 {
diff --git a/core/arch/arm/dts/stm32mp257f-ev1.dts b/core/arch/arm/dts/stm32mp257f-ev1.dts
index 0dcf40ef9..4f0bef74c 100644
--- a/core/arch/arm/dts/stm32mp257f-ev1.dts
+++ b/core/arch/arm/dts/stm32mp257f-ev1.dts
@@ -81,6 +81,7 @@
 		regulators {
 			compatible = "st,stpmic2-regulators";
 
+			ldo1-supply = <&vddio_pmic>;
 			vddcpu: buck1 {
 				regulator-name = "vddcpu";
 				regulator-min-microvolt = <800000>;
@@ -235,6 +236,7 @@
 				regulator-min-microvolt = <1800000>;
 				regulator-max-microvolt = <1800000>;
 				regulator-always-on;
+				st,alternate-input-source;
 				st,mask-reset;
 			};
 			vdd_emmc: ldo2 {
-- 
2.47.2

